File: offer/go/03_3.go
package main

import (
	"fmt"
	"sort"
)

func findRepeatNumber(nums []int) int {
	sort.Ints(nums)
	len := len(nums)
	for i := 0; i < len-1; i++ {
		if nums[i] == nums[i+1] {
			return nums[i]
		}
	}
	return 0
}

func main() {
	nums := []int{2, 3, 1, 0, 2, 5, 3}
	res := findRepeatNumber(nums)
	fmt.Println(res)
}

File: offer/cpp/03_3.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
  int findRepeatNumber(vector<int> &nums) {
    unordered_set<int> set;
    for (const auto &num : nums) {
      if (set.count(num))
        return num;
      set.insert(num);
    }
    return 0;
  }
};

int main(int argc, const char *argv[]) {
  vector<int> nums{2, 3, 1, 0, 2, 5, 3};
  auto res = (new Solution())->findRepeatNumber(nums);
  cout << res << endl;
  return 0;
}